### DISCUSSION TITLE
Any guidance on creating snapshots with vanilla MobX?

### DISCUSSION THREAD

**Discussion by joelbqz** [mobxjs/mobx#2890] [Q&A] [answered]
Hi guys, hope all is well! I know that MST provides an easy way to create/apply snapshots, but I like the simplicity of MobX and wondering what would be the best way when combining multiple stores?

I did a recursive function that saves the entire state tree into a JSON identifying each model with its object
![image](https://user-images.githubusercontent.com/19965325/114444996-2352e680-9b95-11eb-9aa4-c3f3cce287ee.png)

It returns something like this:
```
{
	"__Model": "TodoStore",
	"author": {
		"__Model": "Author",
		"name": "John Doe"
	},
	"todos": [
		{
			"__Model": "Todo",
			"task": ""
		}
	]
}
```
Now I'm looking for a way to apply this to the main store, any idea if `createViewModel` utility would help me on this?
Thanks!!

**Comment by mweststrate** [comment, marked as answer]
you could take a look at serializr or any other serialization library you
can find on the internet. I have to admit in most cases I personally just
handcraft it. Note that your constructor name mechanism will probably not
survive once your code is minified for production. Beyond that building the
inverse utility shouldn't be too hard if you give every class an empty
constructor and register them in some Map by name.

**Reply by joelbqz** [reply]
@mweststrate Thank you! Serializr library definitely looks what I'm looking for

**Reply by jtwigg** [reply]
@joelbqz How was your success with Serializr? I'm hesitant to go for MST or Keystone for the sake a Snapshots...
But Serializr seems to require re-declaration of my type information for the sake of serialization. 

How'd you find this?
